Lib/ssl.py

```diff
@@ -450,6 +451,51 @@ def load_default_certs(self, purpose=Purpose.SERVER_AUTH):
                 self._load_windows_store_certs(storename, purpose)
         self.set_default_verify_paths()
 
+    def load_cert_chain(self, certfile, keyfile=None, password=None):
+       # If `certfile` is bytes or string, treat it as file path.
+       if isinstance(certfile, str) or isinstance(certfile, bytes):
+           certfile_path = certfile
+
+           # If no `keyfile` is given, read private key from `certfile`.
+           if keyfile is None:
+               keyfile_path = certfile_path
+           else:
+               # If `certfile` is bytes or string, expect `keyfile` to be
+               # a bytes or string file path, too.
+               keyfile_path = keyfile
+
+           # Pre CPython 3.7 behavior: let OpenSSL consume the files via
+           # SSL_CTX_use_certificate_chain_file().
```

Comment:
Why not read the files in Python and fall through below? This would allow deleting a bunch of C code.